Modules/Poll/classes/class.ilPollBlockGUI.php

```diff
@@ -96,48 +94,43 @@ public function executeCommand()
 
     public function fillRow($a_poll)
     {
-        $ilCtrl = $this->ctrl;
-        $lng = $this->lng;
-        $ilUser = $this->user;
-        $tpl = $this->main_tpl;
-
-
         // handle messages
 
-        $mess = $this->poll_block->getMessage($ilUser->getId());
+        $mess = $this->poll_block->getMessage($this->user->getId());
         if ($mess) {
             $this->tpl->setVariable("TXT_QUESTION", $mess);
             return;
         }
 
 
         // nested form problem
-        if (!$_SESSION["il_cont_admin_panel"]) {
+        if (!ilSession::get('il_cont_admin_panel')) {
             // vote
 
-            if ($this->poll_block->mayVote($ilUser->getId())) {
+            if ($this->poll_block->mayVote($this->user->getId())) {
                 $this->tpl->setCurrentBlock("mode_info_bl");
                 if ($this->poll_block->getPoll()->getNonAnonymous()) {
-                    $mode_info = $lng->txt("poll_non_anonymous_warning");
+                    $mode_info = $this->lng->txt("poll_non_anonymous_warning");
                 } else {
-                    $mode_info = $lng->txt("poll_anonymous_warning");
+                    $mode_info = $this->lng->txt("poll_anonymous_warning");
                 }
                 $this->tpl->setVariable("MODE_INFO", $mode_info);
                 $this->tpl->parseCurrentBlock();
 
                 $is_multi_answer = ($this->poll_block->getPoll()->getMaxNumberOfAnswers() > 1);
 
-                if (isset($_SESSION["last_poll_vote"][$this->poll_block->getPoll()->getId()])) {
-                    $last_vote = $_SESSION["last_poll_vote"][$this->poll_block->getPoll()->getId()];
-                    unset($_SESSION["last_poll_vote"][$this->poll_block->getPoll()->getId()]);
+                $last_poll_vote = ilSession::get('last_poll_vote');
+                if (isset($last_poll_vote[$this->poll_block->getPoll()->getId()])) {
+                    $last_vote = $last_poll_vote[$this->poll_block->getPoll()->getId()];
+                    unset($_SESSION['last_poll_vote'][$this->poll_block->getPoll()->getId()]);
```

Comment:
To avoid the $_SESSION access, you could do something like

$session_last_poll_vote = ilSession::get('last_poll_vote');
unset($session_last_poll_vote[$this->poll_block->getPoll()->getId()]);
ilSession::set('last_poll_vote, $session_last_poll_vote);